pkg/frontend/querymiddleware/cardinality.go

```diff
@@ -64,8 +64,7 @@ func newCardinalityEstimationMiddleware(cache cache.Cache, logger log.Logger, re
 // Do injects a cardinality estimate into the query hints (if available) and
 // caches the actual cardinality observed for this query.
 func (c *cardinalityEstimation) Do(ctx context.Context, request Request) (Response, error) {
-	spanLog, ctx := spanlogger.NewWithLogger(ctx, c.logger, "cardinalityEstimation.Do")
```

Comment:
Now that cardinality estimation doesn't have it's own span, should the events being logged have some sort of prefix to indicate they're cardinality estimation?

Reply:
According to my analysis they will appear in the parent span which is named `cardinality_estimation`.